tests/unit/test_charm.py

```diff
@@ -29,74 +29,78 @@ def harness():
         harness.cleanup()
 
 
+@pytest.fixture(autouse=True)
+def ceph_conf_directory(tmp_path):
+    with mock.patch("charm.ceph_config_dir") as mock_path:
+        mock_path.return_value = tmp_path / "ceph-conf"
+        yield mock_path
+
+
 @pytest.fixture(autouse=True)
 def mock_apt():
     with mock.patch("charms.operator_libs_linux.v0.apt.add_package", autospec=True) as mock_apt:
         yield mock_apt
 
 
-def test_write_ceph_cli_config(harness):
+@mock.patch("charm.ceph_config_file")
+def test_write_ceph_cli_config(ceph_config_file, harness, ceph_conf_directory):
     """Test writing of Ceph CLI config"""
     harness.begin()
     harness.charm.stored.ceph_data["auth"] = "cephx"
     harness.charm.stored.ceph_data["mon_hosts"] = ["10.0.0.1", "10.0.0.2"]
-    with mock.patch("charm.Path") as mock_path:
-        harness.charm.write_ceph_cli_config()
-        mock_path.assert_called_once_with("/etc/ceph/ceph.conf")
-        path = mock_path.return_value
-        path.open.assert_called_once_with("w")
-        with path.open() as fp:
-            lines = [
-                "[global]",
-                "auth cluster required = cephx",
-                "auth service required = cephx",
-                "auth client required = cephx",
-                "keyring = /etc/ceph/$cluster.$name.keyring",
-                "mon host = 10.0.0.1 10.0.0.2",
-                "log to syslog = true",
-                "err to syslog = true",
-                "clog to syslog = true",
-                "mon cluster log to syslog = true",
-                "debug mon = 1/5",
-                "debug osd = 1/5",
-                "",
-                "[client]",
-                "log file = /var/log/ceph.log",
-                "",
-            ]
-            fp.write.assert_has_calls([mock.call(_l + "\n") for _l in lines])
+    harness.charm.write_ceph_cli_config()
+    path = ceph_config_file.return_value
+    path.open.assert_called_once_with("w")
+    with path.open() as fp:
+        lines = [
+            "[global]",
+            "auth cluster required = cephx",
+            "auth service required = cephx",
+            "auth client required = cephx",
+            f"keyring = {ceph_conf_directory.return_value}/$cluster.$name.keyring",
+            "mon host = 10.0.0.1 10.0.0.2",
+            "log to syslog = true",
+            "err to syslog = true",
+            "clog to syslog = true",
+            "mon cluster log to syslog = true",
+            "debug mon = 1/5",
+            "debug osd = 1/5",
+            "",
+            "[client]",
+            "log file = /var/log/ceph.log",
```

Comment:
It might be a good idea to use a custom logging path in this opportunity something like /var/log/ceph/{app_name}.log. We already have the following logs from other services fwiw.

```
find /var/log/ceph/
/var/log/ceph/
/var/log/ceph/ceph-osd.1.log
/var/log/ceph/ceph.log
/var/log/ceph/ceph-mon.machine-2.log
/var/log/ceph/ceph.audit.log
/var/log/ceph/ceph-mgr.machine-2.log
/var/log/ceph/ceph-volume.log
```

Reply:
Done...

```
/var/log/ceph/ceph-csi-0.log
```

Reply:
uh, it was an easy charm variable to access.  
I'm used to looking at charm logs like:

```
/var/log/juju/unit-<unit>.log
```

so just matched that